build/pr-pipeline.yml

```diff
@@ -64,7 +65,13 @@ stages:
         azurePowerShellVersion: latestVersion
         ScriptType: inlineScript
         Inline: |
-          New-AzResourceGroup -Name "$(DeploymentEnvironmentName)" -Location "$(ResourceGroupRegion)" -Force
+          try
+          {
+            Get-AzResourceGroup -Name $(ResourceGroupName) | Remove-AzResourceGroup -Verbose -Force
+          }
+          catch
+          {}
+          New-AzResourceGroup -Name "$(ResourceGroupName)" -Location "$(ResourceGroupRegion)" -Force
```

Comment:
>New-AzResourceGroup [](start = 10, length = 19)

Aren't this operation idempotent? Why we want drop resource group and everything in it?

Reply:
If the previous build didn't successfully complete the resource group from the last build will still be around. Before this change the old resources would be overwritten by the new ones since they would have the same name. Now new resources with different names will be added. To avoid pilling up a lot of resources in the resource group I added this to clean it up before the new resources are made.

Reply:
Make sense, on the same note, what happens to resources if someone closes PR for failed build? Are they automatically cleaned?

---
In reply to: [540447194](https://github.com/microsoft/fhir-server/pull/1509#discussion_r540447194) [](ancestors = 540447194)

Reply:
This keeps changing though, e.g.
If I have `1.0.1-pr1.1`
I add a commit to fix my build => `1.0.1-pr1.2`
Someone checks into master then I add a commit => `1.0.2-pr1.2`

Reply:
If someone closes a PR with a failed build the resource group is not cleaned up. This is an open issue for us. I had added a script to cleanup a resource group when a PR was closed, but it wasn't working because the GithubAction  to run AzureDevOps pipelines doesn't have a feature we need (https://github.com/Azure/pipelines/issues/17)